src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -182,31 +182,46 @@ private[internal] trait GlbLubs {
         rest filter (t => !first.typeSymbol.isSubClass(t.typeSymbol)))
   }
 
+  /** From a list of types, retain only maximal types as determined by the partial order `po`. */
+  private def maxTypes(ts: List[Type])(po: (Type, Type) => Boolean): List[Type] = {
+    /** Outer loop: fold over `ts` to calculate maximal types, preserving the original order. */
+    @tailrec def outer(ts: List[Type], max: List[Type], rev: Boolean): List[Type] = ts match {
+      case t1 :: ts1 =>
+        /** Inner loop: combine `t1` and the maximal list `max` into the next maximal list. */
+        @tailrec def inner(work: List[Type], done: List[Type]): List[Type] = work match {
+          case t2 :: ts2 if po(t1, t2) => // found a bigger type, we're done here
+            if (!t2.exists(_.isWildcard) || !po(t2, t1)) max
+            else t1 :: (ts2 reverse_::: done)
```

Comment:
Could also do `ts2 reverse_::: t1 :: done` here, but it doesn't seem to matter.

Reply:
Huh, I guess it does matter after all - so frustrating that `with` is not symmetric.